pkgs/by-name/cu/cups-idprt/package.nix

```
@@ -0,0 +1,48 @@
+{ stdenvNoCC
+, lib
+, fetchurl
+, autoPatchelfHook
+, cups
+, unzip
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "cups-idprt";
+  version = "1.4.2";
+
+  src = fetchurl {
+    name = "idprt_tspl_printer_linux_driver_v${version}.zip";
```

Use `finaAttrs` instead of `rec`

```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
  pname = "cups-idprt";
  version = "1.4.2";

  src = fetchurl {
    name = "idprt_tspl_printer_linux_driver_v${finalAttrs.version}.zip";
```

(+extra parenthesis at the end)

---

Is this a requirement? I find the `rec` syntax to be both more intuitive for newbies and generally easier to read. I recognize it can be a potential source of footguns, but I'd strongly prefer to keep this as `rec` if possible.

Obviously, if it's a hard requirement I will switch to the `finalAttrs` syntax.

---

The `rec` syntax works at the language level, meaning that `.override` changing `version` would not update the `name` field in the `fetchurl`

---

> The `rec` syntax works at the language level, meaning that `.override` changing `version` would not update the `name` field in the `fetchurl`

See: https://github.com/NixOS/nixpkgs/issues/315337

---

That's the desired behaviour though? That fetch specifically fetches version `1.4.2`. If a user overrides the `version`, the name of the `fetchurl` derivation _shouldn't_ change because it's still downloading the same `idprt_tspl_printer_linux_driver_v1.4.2.zip` file (not my name; that's iDPRT's name).

---

I've resolved this by just removing the `rec` altogether.